tests/integration/pilot/common/routing.go

```diff
@@ -96,13 +96,18 @@ spec:
       request:
         add:
           istio-custom-header: user-defined-value`,
-				call: func() (echoclient.ParsedResponses, error) {
-					return podA.Call(echo.CallOptions{Target: apps.PodB[0], PortName: "http", Count: callCount})
-				},
-				validator: func(response echoclient.ParsedResponses) error {
-					return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
-						return ExpectString(response.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
-					})
+				call: podA.CallWithRetryOrFail,
+				opts: echo.CallOptions{
+					Target:   apps.PodB[0],
+					PortName: "http",
+					Count:    callCount,
+					Validator: echo.ValidatorFunc(
```

Comment:
The change here seems to be strictly more complicated from a test writer perspective. What are we gaining from the changes here?

Reply:
Hmm seems like if we are just doing `CheckOk` the UX is good but custom stuff is a bit more complicated?

Reply:
Yeah, custom stuff is a little more complicated ... mostly in the cases of chaining multiple validators together. The API may improve a bit if we use the type alias as you suggested above.

Reply:
The API has improved a bit marking resolved for now.